TestApp.Silverlight/TestApp.Silverlight/MainPage.xaml.cs

```diff
@@ -0,0 +1,339 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Linq;
+using System.Net;
+using System.Windows;
+using System.Windows.Browser;
+using System.Windows.Controls;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Animation;
+using System.Windows.Shapes;
+using XamlAnimatedGif;
+
+namespace TestApp.Silverlight
+{
+    public partial class MainPage : INotifyPropertyChanged
+    {
+        public MainPage()
+        {
+            InitializeComponent();
+            _images = new ObservableCollection<string>
+                      {
+                          "images/working.gif",
+                          "images/earth.gif",
+                          "images/radar.gif",
+                          "images/bomb.gif",
+                          "images/bomb-once.gif",
+                          "images/nonanimated.gif",
+                          "images/monster.gif",
+                          "/TestApp.Silverlight;component/images/resource.gif",
+                          "images/partialfirstframe.gif",
+                          "images/newton-cradle.gif",
+                          "images/not-a-gif.png",
+                          "images/optimized-full-code-table.gif",
+                          "http://localhost:33113/images/siteoforigin.gif"
+                          // cross-domain issues
```

Comment:
Correct. If you were making a web request to a domain other than the Silverlight host, the request fails unless there is a security policy file at the root of the web server (see clientaccesspolicy.xml or crossdomain.xml).